### PR TITLE
Implement generate stream for SQL transformer

### PR DESCRIPTION
**Pull request by gatear**
Implementation supports both normal and batched generation.

### CODE REVIEW

**Comment by gatear** [comment]
@snuyanzin it's ready for review. This work was another critical item for my use-case with Spark. 

After this change I'll follow-up on this https://github.com/datafaker-net/datafaker/pull/1261#issuecomment-2172615830 
and show some useful examples for Spark.

Thanks 🙏

**Comment by bodiam** [comment]
@gatear This PR looks good to me, thank you for that, I think this should land in 2.3.0

Anyone else feedback?

**Comment by kingthorin** [comment]
It looks okay to me, but I honestly haven't worked with or on the Transformers at all.

File: src/main/java/net/datafaker/transformations/sql/SqlTransformer.java [hunk 1]
```diff
@@ -394,6 +394,63 @@ private String generateBatchModeStatements(Schema<IN, ?> schema, List<IN> inputs
...
 
+    /**
+     * Represents an interval of integers.
+     * Useful to capture range of rows corresponding to batch SQL statements.
+     *
+     * @see SqlTransformer#generateStream(Schema, long)
+     */
+    private static class Interval {
+        Integer start;
+        Integer end;
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/transformations/sql/SqlTransformer.java:405]:
```suggestion
        final Integer start;
        final Integer end;
```
Is there a reason to have it mot immutable?

**Comment by snuyanzin** [comment]
lgtm, i left a minor comment

**Review by snuyanzin — APPROVED** [review]
Thanks for the contribution and addressing feedback

**Pull request merged**